diff --git a/node_modules/@greenwood/plugin-adapter-vercel/src/index.js b/node_modules/@greenwood/plugin-adapter-vercel/src/index.js
index 407ebc9..cf23897 100644
--- a/node_modules/@greenwood/plugin-adapter-vercel/src/index.js
+++ b/node_modules/@greenwood/plugin-adapter-vercel/src/index.js
@@ -5,17 +5,27 @@ import { checkResourceExists } from "@greenwood/cli/src/lib/resource-utils.js";
 const DEFAULT_RUNTIME = "nodejs20.x";
 
 // https://vercel.com/docs/functions/serverless-functions/runtimes/node-js#node.js-helpers
-function generateOutputFormat(id, type) {
+function generateOutputFormat(id, type, segmentKey) {
   const handlerAlias = "$handler";
   const path = type === "page" ? `${id}.route` : id;
+  // TODO: we should try and support bundling with `URLSearchParms`
+  const extractProps = segmentKey
+    ? type === 'page'
+      ? `const props = { ${segmentKey}: url.split('/')[url.split('/').length - 2] }`
+      : `const props = { ${segmentKey}: url.split('?')[0].split('/').pop() }`
+    : 'const props = {}';
 
   return `
     import { handler as ${handlerAlias} } from './${path}.js';
 
     export default async function handler (request, response) {
-      const { body, url, headers = {}, method } = request;
+      const { body, url, headers = {}, method, query } = request;
+      console.log({ body, url, headers, method });
+      console.log(query);
       const contentType = headers['content-type'] || '';
+      ${extractProps}
       let format = body;
+      console.log({ props });
 
       if (['GET', 'HEAD'].includes(method.toUpperCase())) {
         format = null
@@ -40,7 +50,9 @@ function generateOutputFormat(id, type) {
         headers: new Headers(headers),
         method
       });
-      const res = await ${handlerAlias}(req);
+      const res = '${type}' === 'page'
+        ? await ${handlerAlias}({ request: req, props: {...props} })
+        : await ${handlerAlias}(req, { props });
 
       res.headers.forEach((value, key) => {
         response.setHeader(key, value);
@@ -51,8 +63,8 @@ function generateOutputFormat(id, type) {
   `;
 }
 
-async function setupFunctionBuildFolder(id, outputType, outputRoot, runtime) {
-  const outputFormat = generateOutputFormat(id, outputType);
+async function setupFunctionBuildFolder(id, outputType, outputRoot, runtime, key) {
+  const outputFormat = generateOutputFormat(id, outputType, key);
 
   await fs.mkdir(outputRoot, { recursive: true });
   await fs.writeFile(new URL("./index.js", outputRoot), outputFormat);
@@ -79,22 +91,17 @@ async function vercelAdapter(compilation, options) {
   const adapterOutputUrl = new URL("./.vercel/output/functions/", projectDirectory);
   const ssrPages = compilation.graph.filter((page) => page.isSSR);
   const apiRoutes = compilation.manifest.apis;
+  const dynamicRoutes = [];
 
   if (await checkResourceExists(adapterOutputUrl)) {
     await fs.rm(adapterOutputUrl, { recursive: true });
   }
 
   await fs.mkdir(adapterOutputUrl, { recursive: true });
-  await fs.writeFile(
-    new URL("./.vercel/output/config.json", projectDirectory),
-    JSON.stringify({
-      version: 3,
-    }),
-  );
 
   for (const page of ssrPages) {
     const outputType = "page";
-    const { id, outputHref, route } = page;
+    const { id, outputHref, route, segment } = page;
     // chop off the last / in route, and just use the id if the index route
     const name = id === "index" ? id : `.${route.slice(0, -1)}`;
     const outputRoot = new URL(`${name}.func/`, adapterOutputUrl);
@@ -102,7 +109,7 @@ async function vercelAdapter(compilation, options) {
       (file) => file.startsWith(`${id}.route.chunk`) && file.endsWith(".js"),
     );
 
-    await setupFunctionBuildFolder(id, outputType, outputRoot, runtime);
+    await setupFunctionBuildFolder(id, outputType, outputRoot, runtime, segment?.key);
 
     // handle user's actual route entry file
     await fs.cp(
@@ -117,15 +124,27 @@ async function vercelAdapter(compilation, options) {
         recursive: true,
       });
     }
+
+    if(segment) {
+      const { key } = segment;
+      const basePath = route.replace(`[${key}]/`, '');
+      console.log({ route });
+
+      dynamicRoutes.push({
+        src: `${basePath}(?<${key}>[^/]+)/`,
+        dest: `${basePath}[${key}]`
+      })
+    }
   }
 
   for (const [key, value] of apiRoutes.entries()) {
     const outputType = "api";
-    const { id, outputHref, route } = apiRoutes.get(key);
+    const { id, outputHref, route, segment } = apiRoutes.get(key);
+    console.log({ segment });
     const outputRoot = new URL(`.${route}.func/`, adapterOutputUrl);
     const { assets = [] } = value;
 
-    await setupFunctionBuildFolder(id, outputType, outputRoot, runtime);
+    await setupFunctionBuildFolder(id, outputType, outputRoot, runtime, segment?.key);
 
     await fs.cp(new URL(outputHref), new URL(`./${id}.js`, outputRoot), { recursive: true });
 
@@ -134,8 +153,28 @@ async function vercelAdapter(compilation, options) {
 
       await fs.cp(new URL(asset), new URL(`./${name}`, outputRoot), { recursive: true });
     }
+
+    if(segment) {
+      const { key } = segment;
+      const basePath = route.replace(`[${key}]`, '');
+      console.log({ route });
+
+      dynamicRoutes.push({
+        src: `${basePath}(?<${key}>[^/]+)`,
+        dest: `${basePath}[${key}]`
+      })
+    }
   }
 
+  console.log({ dynamicRoutes });
+  await fs.writeFile(
+    new URL("./.vercel/output/config.json", projectDirectory),
+    JSON.stringify({
+      version: 3,
+      routes: dynamicRoutes
+    }),
+  );
+
   // static assets / build
   await fs.cp(outputDir, new URL("./.vercel/output/static/", projectDirectory), {
     recursive: true,
